query-engine/black-box-tests/tests/metrics/smoke_tests.rs

```diff
@@ -0,0 +1,65 @@
+use query_engine_tests::*;
+
+/// Asserts common basics for composite type writes.
+#[test_suite(schema(schema))]
```

Comment:
I understand it was the easiest way for now, but the query_tests_setup / query_test_macros system is something we will have to overhaul for sustainability.

Reply:
Eventually.

Reply:
You are totally right. I tried to express the same concern, more implicitly in the following statement:

> This paves the path towards addressing another tech-debt point: testing the binary engine as a black box. It's far from being perfect, but as we add more tests we will find opportunities to come up with refinements of the test harness.

Reply:
Yep no worries, I also think we have to incrementally move in the right direction, definitely not a CTA for now